lib/app/features/chat/recent_chats/views/components/recent_chat_tile/recent_chat_tile.dart

```diff
@@ -0,0 +1,250 @@
+                    children: [
+                      Row(
+                        mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                        crossAxisAlignment: CrossAxisAlignment.end,
+                        children: [
+                          SenderSummary(chat: chat),
+                          ChatTimestamp(chat: chat),
+                        ],
+                      ),
+                      SizedBox(height: 2.0.s),
+                      Row(
+                        mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                        children: [
+                          ChatPreview(chat: chat),
+                          UnreadCountBadge(unreadCount: chat.unreadMessageCount),
+                        ],
+                      ),
+                    ],
+                  ),
+                ),
+              ],
+            ),
+          ),
+        ),
+      ],
+    );
+  }
+}
+
+class SenderSummary extends StatelessWidget {
+  const SenderSummary({required this.chat, super.key});
+  final RecentChatDataModel chat;
+
+  @override
+  Widget build(BuildContext context) {
+    return Row(
+      children: [
+        Text(
+          chat.sender.name,
+          style: context.theme.appTextThemes.subtitle3,
+        ),
+        if (chat.sender.isIceUser)
+          Padding(
+            padding: EdgeInsets.only(left: 4.0.s),
+            child: Assets.svg.iconBadgeIcelogo.icon(size: 16.0.s),
+          ),
+        if (chat.sender.isApproved)
+          Padding(
+            padding: EdgeInsets.only(left: 4.0.s),
+            child: Assets.svg.iconBadgeVerify.icon(size: 16.0.s),
+          ),
+      ],
+    );
+  }
+}
+
+class ChatTimestamp extends StatelessWidget {
+  const ChatTimestamp({
+    required this.chat,
+    super.key,
+  });
+
+  final RecentChatDataModel chat;
+
+  @override
+  Widget build(BuildContext context) {
+    return Text(
+      formatMessageTimestamp(chat.message.time),
+      style: context.theme.appTextThemes.body2.copyWith(
+        color: context.theme.appColors.onTertararyBackground,
+      ),
+    );
+  }
+}
+
+class ChatPreview extends StatelessWidget {
+  const ChatPreview({required this.chat, super.key});
+  final RecentChatDataModel chat;
+
+  @override
+  Widget build(BuildContext context) {
+    return Expanded(
+      child: Row(
+        children: [
+          RecentChatMessageIcon(message: chat.message),
+          Expanded(
+            child: Text(
+              maxLines: 2,
+              overflow: TextOverflow.ellipsis,
+              _getMessageContentText(chat.message, context),
+              style: context.theme.appTextThemes.body2.copyWith(
+                color: context.theme.appColors.onTertararyBackground,
+              ),
+            ),
+          ),
+        ],
+      ),
+    );
+  }
+
+  String _getMessageContentText(RecentChatMessage message, BuildContext context) {
+    switch (message) {
```

Comment:
Could you please change it to `switch-expression` for better readability?

Reply:
done